### DISCUSSION TITLE
Quartz.NET 3.8.0

### DISCUSSION THREAD

**Discussion by lahma** [quartznet/quartznet#2198] [Announcements]
This most notably tries to improve Quartz API and metadata to handle NET 8 trimming capabilities better, but there's probably still issues to encounter. When using trimming one should use the strongly-typed MS DI integration and `SchedulerBuilder` APIs.

## What's Changed
* Add Qodana scanning by @lahma in https://github.com/quartznet/quartznet/pull/2110
* Ensure that the DailyCalendar use the same TimeZone offset for all the checks by @kvpt in https://github.com/quartznet/quartznet/pull/2113
* Fix for JobDataMap, TryGetValue return false when key is null/doesn't exist by @jafin in https://github.com/quartznet/quartznet/pull/2125
* SendMailJob to throw JobExecutionException on BuildMessage construction failure due to missing mandatory params. by @jafin in https://github.com/quartznet/quartznet/pull/2126
* Backport test changes to  ReschedulingTriggerShouldKeepOriginalNextFireTime from main to 3.x by @jafin in https://github.com/quartznet/quartznet/pull/2129
* JobInterruptMonitorPlugin should tolerate missing JobDataMapKeyAutoInterruptable by @lahma in https://github.com/quartznet/quartznet/pull/2191
* Add NET 8.0 targeting by @lahma in https://github.com/quartznet/quartznet/pull/2192
* Upgrade TimeZoneConverter to version 6.1.0 by @lahma in https://github.com/quartznet/quartznet/pull/2194
* Improve trimming compatibility by @lahma in https://github.com/quartznet/quartznet/pull/2195
* Support injecting dependencies to XMLSchedulingDataProcessorPlugin by @lahma in https://github.com/quartznet/quartznet/pull/2196
* Enable trimmed publish for ASP.NET Core example by @lahma in https://github.com/quartznet/quartznet/pull/2197

## New Contributors
* @kvpt made their first contribution in https://github.com/quartznet/quartznet/pull/2113

**Full Changelog**: https://github.com/quartznet/quartznet/compare/v3.7.0...v3.8.0

<hr /><em>This discussion was created from the release <a href='https://github.com/quartznet/quartznet/releases/tag/v3.8.0'>Quartz.NET 3.8.0</a>.</em>

**Comment by cn-byn** [comment]
While I welcome the changes to TryGetValue, the changelog should mention that this change is not only "fixing" the TryX methods, but also affecting the GetX methods in a breaking manner.

We have to go through all of our jobs now and look at every usage of GetX because, for example, GetBoolean now might throw where before it just returned false.
